packages/core/src/State.ts

```diff
@@ -298,7 +323,9 @@ export class State<
    * @param parentStateValue
    */
   public matches<TSV extends TTypestate['value']>(
-    parentStateValue: TSV
+    parentStateValue: TResolvedTypesMeta extends TypegenEnabled
+      ? Prop<TResolvedTypesMeta, 'matchesStates'>
```

Comment:
Nit: wondering if a utility `PropOrDefault<TResolvedTypesMeta, 'matchesStates', TSV>` would be useful for DRYing some of the types